src/config.ml

```diff
@@ -1108,7 +1122,7 @@ let valid_server_options ~client:_ _server_t =
 
 let parse_next (effect:parser_effect) initial_state : (parser_state, 'err) result =
   let open Rresult in
-  let rec loop (acc:Conf_map.t) : line list -> (parser_state,'b) result =
+  let rec loop (acc:Conf_map.t) rport : line list -> (parser_state,'b) result =
```

Comment:
I'm a little bit skeptical of adding a parameter for `rport` to this function.
I think a better approach may be to make the lexer for `remote` register a polymorphic variant `Default_port`, and rewriting in the end (before return to the consumer) to be able to correctly parse:
```
remote foo.example 1194
rport 1234
remote bar.example
```
which should result in
```
Remote foo.example 1194
Remote bar.example 1234
```
I think this implementation (unless I'm reading it wrong) would result in
```
Remote foo.example 1234
Remote bar.example 1234
```

(idk, maybe that would be a good test case to add). In any case I'm okay with merging this PR as-is.

Reply:
> I think a better approach may be to make the lexer for remote register a polymorphic variant `Default_port`, and rewriting in the end (before return to the consumer)

I'd be in favour of keeping the key-value of `Remote` as is in our config GADT. I thought loop is exactly the right place to transform other line directives to configuration map elements. I don't quire understand what "in the end" in your sentence should mean.